a3p-integration/proposals/z:acceptance/test-lib/auction-lib.js

```diff
@@ -0,0 +1,305 @@
+import {
+  addPreexistingOracles,
+  agd,
+  agopsInter,
+  agoric,
+  ATOM_DENOM,
+  CHAINID,
+  executeOffer,
+  getPriceQuote,
+  GOV1ADDR,
+  pushPrices,
+  VALIDATORADDR,
+} from '@agoric/synthetic-chain';
+import {
+  retryUntilCondition,
+  waitUntilAccountFunded,
+  waitUntilOfferResult,
+} from './sync-tools.js';
+import { boardSlottingMarshaller, makeFromBoard } from './rpc.js';
+import { AmountMath } from '@agoric/ertp';
+
+/**
+ * Typo will be fixed with https://github.com/Agoric/agoric-sdk/pull/10171
+ * @typedef {import('./sync-tools.js').RetyrOptions} RetryOptions
+ */
+
+const ambientAuthority = {
+  query: agd.query,
+  follow: agoric.follow,
+  setTimeout: globalThis.setTimeout,
+};
+
+export const scale6 = x => BigInt(x * 1_000_000);
+
+const fromBoard = makeFromBoard();
+const marshaller = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+
+// Import from synthetic-chain once it is updated
+export const bankSend = (from, addr, wanted) => {
+  const chain = ['--chain-id', CHAINID];
+  const fromArg = ['--from', from];
+  const testKeyring = ['--keyring-backend', 'test'];
+  const noise = [...fromArg, ...chain, ...testKeyring, '--yes'];
+
+  return agd.tx('bank', 'send', from, addr, wanted, ...noise);
+};
+
+export const pushPricesForAuction = async (t, price) => {
+  const oraclesByBrand = new Map();
+  await addPreexistingOracles('ATOM', oraclesByBrand);
+
+  await pushPrices(price, 'ATOM', oraclesByBrand, t.context.roundId + 1);
+
+  await retryUntilCondition(
+    () => getPriceQuote('ATOM'),
+    res => res === `+${scale6(price).toString()}`,
+    'price not pushed yet',
+    {
+      log: t.log,
+      setTimeout: globalThis.setTimeout,
+      ...t.context.pushPriceRetryOpts,
+    },
+  );
+};
+
+/**
+ * @param {any} t
+ * @param {{
+ *  name: string
+ *  offerId: string,
+ *  depositValue: string,
+ * }} depositor
+ * @param {Object<string, {
+ *  bidder: string,
+ *  bidderFund: {
+ *   value: number,
+ *   denom: string
+ *  },
+ *  offerId: string,
+ *  give: string,
+ *  price?: number
+ *  discount?: string
+ * }>} bidders
+ */
```

Comment:
one-space indentation is insufficient. Please use two spaces.

Reply:
Resolved in https://github.com/Agoric/agoric-sdk/commit/bf054144ec43667195d237a0f72f3cb1760e4ff9